### PR TITLE
[7.17.1] Adjust indices.recovery.max_bytes_per_sec according to external settings

### PR DESCRIPTION
**Pull request by tlrx**
This pull request backports the node bandwidth settings merged in 8.1.0.

It cherry-picks the following changes:
- #82819
- #83350
- #83359
- #83361

The commit e235fea6e407314c38c61c104f6fe6161c0e87b4 contains the changes for 7.17.1:
- the user factors  settings `node.bandwidth.recovery.factor.read` and `node.bandwidth.recovery.factor.write` are removed to only keep the operator variants set by the platform
- the operator variants are made static (non dynamic and non-operator only)
- a warning in the documentation to warn about the recovery settings not being available in 8.0.0

### CODE REVIEW

File: docs/changelog/82819.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+pr: 82819
+summary: "[Draft] Adjust `indices.recovery.max_bytes_per_sec` according to external\
```
Comment (henningandersen) [inline docs/changelog/82819.yaml:2]: Can we remove `[Draft]` here (and preferably in 8.1 too)?

File: docs/changelog/83350.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+pr: 83350
+summary: Add missing max overcommit factor to list of (dynamic) settings
+area: Recovery
+type: bug
```
Comment (henningandersen) [inline docs/changelog/83350.yaml:4]: Perhaps we can remove this changelog entry?

**Review by henningandersen — APPROVED** [review]
LGTM.

File: docs/changelog/82819.yaml (hunk 1 above)
Comment (tlrx) [reply docs/changelog/82819.yaml:2]: I opened #83527 for 8.1, thanks for catching this

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Comment by tlrx** [comment]
Thanks Henning and David

**Pull request merged**